source/_posts/2017-02-07-falsehoods-programmers-believe-about-language.markdown

```diff
@@ -0,0 +1,189 @@
+something like `"#{user.name} is in the #{user.location}."` where the user's
+name is, well, their name like "Ben Hamill" and the location might be one of
+"yard", "kitchen", "bathroom" or "bedroom". That seems reletively
+straightforward... in English.
+
+In languages with gendered nouns, those four location words might have different
+genders, meaning that the untemplated "the" would be different for different
+locations. If the translator lacks the context for what might go into that slot,
+they'll have no hope of being right all the time. For instance, in Spanish, "the
+kitchen" is "la cocina", but "the bathroom" is "el baño". If the translator has
+to pick either "el" _or_ "la", they're guaranteed to be wrong some of the time.
+
+
+## If I make sure to group things like articles in with nouns, that'll sort gendered languages.
+
+If you change the above example to `"#{user.name} is in #{user.location}."` and
+then make the location options "the yard", "the kitchen", "the bathroom" or "the
+bedroom", you _will_ sort out that particular sentence (as far as I know, but
+see below). But consider something like `"Your friend is wearing a #{garmet.color} #{garmet.type}."`
+Many languages expect adjectives to agree on gender with the noun they modify.
+If your friend could be wearing a shirt or a hat that can be red or purple, you
+need two of each color and need to be aware of which garmet needs which version
+of the appropriate color.
+
+
+## I can write enough grammatical metadata to be able to stitch togeter coherent sentences in all the languages I'm targeting.
+
+There are whole research projects by leading linguists and computer scientists
+working on this problem. Solving this in general is _ridiculously_ hard, it
+seems. Even solving it for one specific pair of languages requires you to know a
+ton about the grammatical rules of both languages and most native English
+speakers cannot express the grammar rules they routinely adhere to on the fly as
+they speak, let alone with such rigor that a computer could make use of the
+instructions. Also, the amount of metadata you'd have to have about everything
+that might appear in a template and every template or every slot in every
+template is pretty unwieldy. Unless this is your company's core competency, you
+should probably be solving other problems with your time.
+
+
+## Well, templating is bad, but if I allow for two versions of every English sentence, I'll be OK for gendered languages.
+
+This is wrong in two ways. First, two and one are not the only options for
+number of grammatical genders languages have. Some languages have masculine,
+feminine and _neuter_ nouns like German, Icelandic, Russian, and Polish. And
+some languages (with, I'll admit, reletively smaller speaking communities) have
+even more like Czech, which has four. Wikipedia, of course, has a
+[big, categorized list](https://en.wikipedia.org/wiki/List_of_languages_by_type_of_grammatical_genders).
```

Comment:
Woah, to hell with Shona. What even? On the other hand, maybe porting their ideas would solve some of our grammar vs gender identity mishaps.